File: D3/2805.cpp
// 제목 : 농작물 수확하기
// 링크 : https://swexpertacademy.com/main/code/problem/problemDetail.do?contestProbId=AV7GLXqKAWYDFAXB
// 메모리 :  12,680 kb
// 실행시간 : 9 ms

#include<iostream>
#include<cstdio>
#include<cstdlib>

using namespace std;

int main(int argc, char** argv)
{
    int test_case;
    //테스트 케이스 수
    int T;
    cin >> T;
    for (test_case = 1; test_case <= T; ++test_case)
    {
        //농장 크기 입력받기
        int n;
        scanf("%d", &n);

        //농장을 2차원 배열로 만들고, 숫자 하나씩 끊어서 정보를 입력받음
        int** map = new int* [n];
        for (int i = 0; i < n; i++) {
            map[i] = new int[n];
            for (int j = 0; j < n; j++) {
                scanf("%1d", &map[i][j]);
            }
        }

        int sum = 0;
        int row = n / 2;

        //규칙에 따라 수확하기
        for (int i = 0; i < n; i++) {
            int Row = abs(row);
            for (int j = Row; j < n - Row; j++) {
                sum += map[i][j];
            }
            row--;
        }

        //결과 출력
        printf("#%d %d\n", test_case, sum);


        //배열 동적할당 해제
        for (int i = 0; i < n; i++) {
            delete[] map[i];
        }
        delete[] map;
    }
    return 0;//정상종료시 반드시 0을 리턴해야합니다.
}

File: D3/2805.java
// 제목 : 농작물 수확하기
// 링크 : https://swexpertacademy.com/main/code/problem/problemDetail.do?contestProbId=AV7GLXqKAWYDFAXB
// 메모리 :  20,996 kb
// 실행시간 : 118 ms

import java.io.BufferedReader;
import java.io.InputStreamReader;
  
class Solution
{
	public static void main(String args[]) throws Exception
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        
        //테스트 케이스 수
        int T = Integer.parseInt(br.readLine());
        
        //각각의 테스트 케이스 마다
        for(int test_case = 1; test_case <= T; test_case++)
        {
        	//농장의 크기를 입력받아 배열을 만들었음
        	int n = Integer.parseInt(br.readLine());
        	char[][] arr = new char[n][];
        	
        	//줄별로 정보를 입력받아 배열에 저장함
        	for(int i=0;i<n;i++) {
        		String s = br.readLine();
        		arr[i] = s.toCharArray();
        	}
        	
        	//모양에 맞추어서 농작물 수확
        	int answer=0;
        	for(int i=0;i<n;i++) {
        		int tab = Math.abs(n/2-i);
        		for(int j=tab;j<n-tab;j++) answer+=arr[i][j]-'0';
        	}
        	
        	
        	//결과문 저장
        	sb.append('#').append(test_case).append(' ').append(answer).append('\n');
        }
        //출력
        System.out.printf(sb.toString());
    }
}